src/Changelog.js

```diff
@@ -142,74 +183,122 @@ export default class Changelog {
     return Object.keys(committers).map(k => committers[k]).sort();
   }
 
-  getCommitInfo() {
+  getCommitsInfo() {
     const commits = this.getListOfCommits();
 
     progressBar.init(commits.length);
 
-    var logs = commits.map(commit => {
+    const commitsInfo = commits.map(commit => {
+      // commit is formatted as following:
+      // <short-hash>;<ref-name>;<summary>;<date>
+      const parts = commit.split(";");
+      const sha = parts[0];
+      const _refs = parts[1].split(",");
+      let tag;
+      if (_refs.length === 1) {
+        // "tag: <tag-name>
+        tag = _refs[0].replace(/tag:\s(.*?)$/, "$1").trim();
+      } else if (_refs.length === 4) {
+        // "HEAD -> master, tag: <tag-name>, origin/master, origin/HEAD"
+        tag = _refs[1].replace(/tag:\s(.*?)$/, "$1").trim();
+      }
+      const message = parts[2];
+      const date = parts[3];
```

Comment:
This part has been refactored to parse the different commit info, especially the tag name